lib/ui/home/home_page.dart

```diff
@@ -590,3 +581,83 @@ class ShieldFadeInAnimationCurve extends Curve {
     }
   }
 }
+
+//animated hamburger menu
+class HamburgerMenu extends ConsumerStatefulWidget {
+  // 0 for idle
+  // 1 for upward icon
+  // -1 for back icon
+  final int iconState;
```

Comment:
Could we use an enum here instead? Something like `HamburgerState`?

Reply:
got it.